src/dataGenerator.ts

```diff
@@ -103,77 +48,45 @@ export default async function dataGenerator({
         payload = crypto.randomBytes(global.recordSize).toString('hex');
     }
 
-    let registry;
-    let producer;
-    let avroSchemas = {};
-    if(global.dryRun !== true){
-        producer = await connectKafkaProducer();
+    let producer: KafkaProducer | null = null;
+    if (global.dryRun !== true) {
+        let outputFormat: OutputFormat;
+        if (format === 'avro') {
+            outputFormat = await AvroFormat.create();
+        } else if (format === 'json') {
+            outputFormat = new JsonFormat();
+        }
+
+        producer = await KafkaProducer.create(outputFormat);
     }
+
     for await (const iteration of asyncGenerator(number)) {
         global.iterationIndex = iteration;
         let megaRecord = await generateMegaRecord(schema);
 
         if (iteration == 0) {
-            if (format == 'avro') {
-                if (global.dryRun) {
-                    alert({
-                        type: `success`,
-                        name: `Dry run: Skipping schema registration...`,
-                        msg: ``
-                    });
-                } else {
-                    registry = await schemaRegistryConfig();
-                }
-            }
             for (const topic in megaRecord) {
-                await prepareTopic(topic);
-                if (format == 'avro' && global.dryRun !== true) {
-                    avroSchemas = await prepareSchema(
-                        megaRecord,
-                        topic,
-                        registry,
-                        avroSchemas
-                    );
-                }
+                await producer?.prepare(topic, megaRecord);
             }
         }
 
         for (const topic in megaRecord) {
             for await (const record of megaRecord[topic].records) {
-                let encodedRecord = null;
-                let recordKey = null;
+                let key = null;
                 if (record[megaRecord[topic].key]) {
-                    recordKey = record[megaRecord[topic].key];
+                    key = record[megaRecord[topic].key];
                 }
 
                 if (global.recordSize) {
                     record.recordSizePayload = payload;
                 }
 
-                if (global.dryRun) {
-                    alert({
-                        type: `success`,
-                        name: `Dry run: Skipping record production...`,
-                        msg: `\n  Topic: ${topic} \n  Record key: ${recordKey} \n  Payload: ${JSON.stringify(
```

Comment:
We now don't output records to the terminal during dry run, which is why the tests are failing. I suggest keeping just this `if (global.dryRun) {...}` part

Reply:
I'll re-add it for now, but I want to get rid of those tests soon. Testing based on stdout is really fragile and we have proper integration tests

Reply:
I don’t mind getting rid of the tests, but the user should be able to see what the records look like during dry run regardless